client/state/sites/selectors/get-woo-express-trial-expiration.ts

```diff
@@ -0,0 +1,21 @@
+import { PLAN_ECOMMERCE_TRIAL_MONTHLY } from '@automattic/calypso-products';
+import { getPurchases } from 'calypso/state/purchases/selectors';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Returns the expiration date of the Woo Express trial. If the trial is not active, returns null.
+ *
+ * @param {AppState} state - Global state tree
+ * @returns {string|null}
+ */
+export default function getWooExpressTrialExpiration( state: AppState ) {
+	const trialPurchase = getPurchases( state ).find(
+		( purchase ) => purchase.productSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY
+	);
+
+	if ( ! trialPurchase ) {
+		return null;
+	}
```

Comment:
Can we use the `getCurrentPlan()` selector here and then check whether the current plan is the eCommerce trial?
```suggestion
	// import { getCurrentPlan } from 'client/state/sites/plans/selectors';
	const currentPlan = getCurrentPlan( state, siteId );
	if ( ! currentPlan || ! isECommerceTrialPlan( currentPlan ) ) {
		return null;
	);

	return currentPlan.expiryDate;
```